### PR TITLE
[Run] Ignore `typing` module for input parsing

### PR DESCRIPTION
**Pull request by guy1992l**
`typing` type hints like `Union`, `List` and `Optional` will be ignored (so the object's type will remain a `mlrun.DataItem`) when the auto-parse of the `mlrun.handler` will get them. Once the `mlrun.package` module is implemented, `typing` will be supported.

### CODE REVIEW

File: mlrun/run.py [hunk 1]
```diff
@@ -1274,9 +1274,9 @@ def _parse_type_hint(type_hint: Union[Type, str]) -> Type:
     """
     Parse a given type hint from string to its actual hinted type class object. The string must be one of the following:
 
-    * Python builtin type - one of `tuple`, `list`, `set`, `dict` and `bytearray`.
-    * Full module import path. An alias (if import pandas as pd is used, the type hint cannot be `pd.DataFrame`) is
-      not allowed.
+    * Python builtin type - one of ``tuple``, ``list``, ``set``, ``dict`` and ``bytearray``.
+    * Full module import path. An alias is not allowed (if ``import pandas as pd`` is used, the type hint cannot be
```
Comment (Tankilevitch) [inline mlrun/run.py:1278]: Add example to how should it be, if alias passed

**Review by Tankilevitch — APPROVED** [review]
Looks good minor suggestion

File: docs/concepts/decorators-and-auto-logging.md [hunk 2]
```diff
@@ -87,6 +87,8 @@ def train_and_predict(train_data: pd.DataFrame,
 ...
 ```
 
+**Notice**: `typing` hints like `Optional`, `Union` and `List` are currently not supported but will be in the future.
```
Comment (alonmr) [inline docs/concepts/decorators-and-auto-logging.md:90]: I think a message like the following will make it clearer that the whole typing library is not supported
```suggestion
**Notice**: type hints from the`typing` library (e.g. `typing.Optional`, `typing.Union`, `typing.List` etc.) are currently not supported but will be in the future. 
```

**Review by alonmr — COMMENTED** [review]
LGTM. Minor phrasing suggestion.

**Pull request merged**